packages/pyright-internal/src/common/uri/uri.ts

```diff
@@ -0,0 +1,206 @@
+/*
+ * uri.ts
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ *
+ * URI namespace for storing and manipulating URIs.
+ */
+
+import { URI, Utils } from 'vscode-uri';
+import { EmptyUri } from './emptyUri';
+import { FileUri } from './fileUri';
+//import * as memoization from './memoization_instrumenting';
+import { WebUri } from './webUri';
+
+export interface Uri {
+    // Unique key for storing in maps.
+    readonly key: string;
+
+    // Returns the scheme of the URI.
+    readonly scheme: string;
+
+    // Returns the last segment of the URI, similar to the UNIX basename command.
+    readonly basename: string;
```

Comment:
The term `basename` means nothing to me. Maybe I just need to add this to my internal dictionary, but I wonder if there's a better name.

.NET chose to name their equivalent as [`GetFileName`](https://learn.microsoft.com/en-us/dotnet/api/system.io.path.getfilename?view=net-8.0#system-io-path-getfilename(system-string)). Maybe we should follow their lead?

GetFileName("c:\foo\") returns ""
GetFileName("c:\foo\bar") returns "bar"
GetFileName("c:\foo\bar.txt") returns "bar.txt"

Reply:
filename makes sense to me. I'll change it to that.